lib/diff.ml

```diff
@@ -4,17 +4,26 @@ type ('item, 'diff) t = Added of 'item | Removed of 'item | Modified of 'diff
 
 type 'a atomic_modification = { reference : 'a; current : 'a }
 (** The simplest diff representation for the modification of a value of type 'a.
-     [reference] is the value before and [current] is the value after the change occured.
-     Use this type when there is no better representation available. *)
+    [reference] is the value before and [current] is the value after the change
+    occured. Use this type when there is no better representation available. *)
 
 type value = {
   vname : string;
   vdiff : (value_description, value_description atomic_modification) t;
 }
 
+type 'changed type_modification =
```

Comment:
Is there a specific reason why this is polymorphic?

My guess is that it's to leave room for the variant thing but I think we could still remove the parameter as we'll likely have to use some sort of:
```
| Variant of constructor list
| Record_diff of label_ list
...
```

Since the `type_declaration` type itself is not polymorphic, I don't think we need to be here but I might be missing something.

Reply:
Yes, I was considering that for the variant case. I think doing it as the compiler does it might be a possibility too. 
`type_decl_kind = (label_declaration, constructor_declaration) type_kind`

I will think about that once I rebase the variant PR.